apps/remote_control/lib/lexical/remote_control/project_node.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Lexical.RemoteControl.ProjectNode do
+  alias Lexical.RemoteControl
+  require Logger
+
+  defmodule State do
+    defstruct [:project, :paths, :cookie]
+
+    def new(project) do
+      paths = format_prepending_paths(RemoteControl.glob_paths())
+      cookie = Node.get_cookie()
+
+      struct!(__MODULE__, paths: paths, cookie: cookie, project: project)
+    end
+
+    defp format_prepending_paths(paths_as_charlists) do
+      Enum.map_join(paths_as_charlists, " -pa ", &Path.expand/1)
```

Comment:
this is really not a use case for map_join, as you're going to miss the initial `-pa` 
Similarly, the name of "paths" is incorrect, they're not paths, they're actually path append arguments. 
For naming, i suggest keeping the paths as actual binary paths and making the following function:

```elixir
def path_append_arguments(%__MODULE__{} = state) do 
   Enum.map(state.paths, fn path -> 
    expanded_path = Path.expand(path)
     "-pa #{expanded_path} "
   end  
  |> IO.iodata_to_binary()
end
```